frontend/src/lib/components/widgets/ChatInput/ChatInput.tsx

```diff
@@ -0,0 +1,209 @@
+      element.setValue = false
+      setValue(value)
+      setDirty(true)
+    }
+  }, [element])
+
+  useEffect(() => {
+    const scrollHeight = getScrollHeight()
+    setScrollHeight(scrollHeight)
+  }, [value])
+
+  const realHeight = Math.min(Math.max(scrollHeight, MIN_HEIGHT), MAX_HEIGHT)
+
+  return (
+    <StyledChatInputContainer
+      className="stChatInputContainer"
+      width={width}
+      position={element.position}
+    >
+      <StyledChatInput>
+        <UITextArea
+          data-testid="stChatInput"
+          inputRef={chatInputRef}
+          value={value}
+          placeholder={element.placeholder}
+          onChange={handleChange}
+          onKeyDown={handleKeyDown}
+          aria-label={element.placeholder}
+          disabled={element.disabled}
+          rows={1}
+          overrides={{
+            Root: {
+              style: {
+                // Baseweb requires long-hand props, short-hand leads to weird bugs & warnings.
+                borderLeftWidth: 0,
+                borderTopWidth: 0,
+                borderRightWidth: 0,
+                borderBottomWidth: 0,
+                borderLeftStyle: "none",
+                borderTopStyle: "none",
+                borderRightStyle: "none",
+                borderBottomStyle: "none",
+                outline: "none",
+                borderBottomLeftRadius: theme.radii.md,
+                borderBottomRightRadius: theme.radii.none,
+                borderTopLeftRadius: theme.radii.md,
+                borderTopRightRadius: theme.radii.none,
+                backgroundColor: theme.colors.transparent,
+                ":focus-within": {
+                  border: `none`,
+                },
+              },
+            },
+            InputContainer: {
+              style: {
+                backgroundColor: theme.colors.transparent,
+              },
+            },
+            Input: {
+              style: {
+                lineHeight: "1.4",
+                height: `${realHeight}px`,
+                minHeight: `${MIN_HEIGHT - 2}px`,
+                backgroundColor: theme.colors.transparent,
+                "::placeholder": {
+                  opacity: "0.7",
+                },
+                // Baseweb requires long-hand props, short-hand leads to weird bugs & warnings.
+                paddingRight: theme.spacing.sm,
+                paddingLeft: theme.spacing.sm,
+                paddingBottom: theme.spacing.sm,
+                paddingTop: theme.spacing.sm,
+              },
+            },
+          }}
+        />
+        <InputInstructions
+          dirty={dirty}
+          value={value}
+          maxLength={element.maxChars}
+          // TODO: Refactor InputInstructions to be more semantic
+          type="single"
+        />
+      </StyledChatInput>
+      <StyledSendIconButton height={realHeight} onClick={handleSubmit}>
+        <Icon content={Send} size="lg" />
+      </StyledSendIconButton>
+    </StyledChatInputContainer>
```

Comment:
nit: an alternative to adding the white background as a dedicated element could also be to use `::after` on the StyledChatInputContainer